In [198]:
%reload_ext sql

In [199]:
from sqlalchemy import create_engine

In [200]:
%sql postgresql://oceane:LunedeTerre@localhost/postgres

In [201]:
import pandas as pd

In [202]:
dataset = pd.read_csv("titanic_train.csv")

In [30]:
dataset.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [31]:
dataset.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

Activités à réaliser :

-Modèle de donnée d'une base de donnée réel (TITANIC)
-Script creation base de données 
-Manipulation du language sql 
-Exploitation d'une base de donnée et visualisation des résultats

## PHASE 1 ##

Pour faire bien analyser les données nous nous sommes posée les questions suivantes afin de comprendre les différents liens que nous pourrions faire pour exploiter aux mieux ces données :
Quels est le profil des passagers ? Le bateau était majoritairement composé d'hommes ou de femmes  ?

Dans un premier temps, suite aux informations trouvé sur le site « kaggle », nous remarquons que les passagers étaient majoritairement des hommes. Sont-ils jeunes ou agées ?
Nous remarquons qu’il y a des personnes de tout âge confondu, des tranches d’âge
différentes tels que des enfants entre 1 et 12 ans , des adolescents entre 13 et 18 ans... 
Cependant , il y a une grande majorité d’adulte plus ou moins jeunes (ne pas hésiter à regrouper par tranche d’âge) Les personnes ayant embarqué au port de Southampton semblent être les personnes qui ont le
moins survécu (voir ligne 71 à 75 du tableau). Cette base de données Titanic nous donne toutes les informations concernant les individus qui étaient présents au moment du drame.

ANALYSE GLOBAL :

- Le fichier est composé de 12 colonnes (Attributs).
- Le fichier est composé de 891 lignes (Tuples).
- Les différentes données sont sous fichier csv donc séparées/délimités par des points virgules
- Le fichier exploite les données des personnes ayant survécu ou non au drame du Titanic

- Dans ce fichier nous retrouvons les attributs suivants : 

- Le numéro d’identité des passagers
- Les survivants représentés par un 1 et ceux n’ayant pas survécu représentés par un 0.
- La classe d’embarquement (1,2 ou 3).
- Noms des passagers
- Le sexe (H ou F)
- L’Age des personnes présentent dans le bateau
- La situation /relation familiale des passagers (mariés / frères sœurs)
- Parch= Représente le nombre d’enfants et parents à bord du Titanic
- Le numéro des billets
- Tarif des passagers.
- Numéros de cabine des passagers
- Le port dans lequel ils ont embarqués (Cherbourg, Queenstown et Southampton) 

La ville de départ était Southampton en Angleterre et le trajet devait se poursuivre jusqu’à New York donc il y a eu une escale à Cherbourg en France puis à Queenstown en Nouvelle Zélande.
Nous remarquons que certaines données sont manquantes notamment celle concernant le numéro de cabine. 
Certaines données n’ont pas été retranscrite celle-ci sont alors interprété comme étant nul. 




OBJECTIFS DU PROJET : 

Le but est d’exploiter les connaissances que nous allons acquérir dans ce semestre en Bdd afin de manipuler une base de données, créer des tables et répondre à certaines requêtes concernant des données réels.
Parallèlement nous allons également apprendre à faire des requêtes en langage SQL. Cela nous permet également d'avancer en fonction des cours que nous recevons, cette sae nous permet donc de passer de la théorie à la pratique. 

L’objectif secondaire est de déterminer les caractéristiques des personnes qui ont survécu et celle qui n’ont pas survécu. Ses analyses vont donc nous permettre de répondre à des questions beaucoup plus précises afin de
comprendre si certains facteurs sont liés à cette tragédie, au fait que d’autres ont survécu ou
non. Comme par exemple , savoir combien de personnes ont survécu, quelles étaient leurs situations au moment des
faits…



MODELE DE DONNEES TITANIC TRAIN :


titanic_train_csv(passagerId, survived, Pclass, Name, Sex, Age, Sibsp, Parch, Ticket, Fare, Cabin, Embarked) ;

Dom(passagerId) = valeurs numériques
Dom(survived) =valeurs numériques
Dom(Pclass) =valeurs numériques
Dom(Name) = chaine de caractère 
Dom(Sex) =chaine de caractère 
Dom(Age) =float
Dom(Sibsp) =valeur numérique 
Dom(Parch) =valeur numérique 
Dom(Ticket) =valeur numérique 
Dom(Fare) =float
Dom(Cabin)=ensemble des valeurs autorisés
Dom(Embarked) =chaine de caractère

Cardinalité = 891 Degré = 12

## PHASE 2 ##

Objectifs : 

-Creation de tables nécessaire suite à l'analyse faite des données de la base Titanic_Train_csv 
-Description des tables 
-Alimention des tables à partir du fichier


Description des tables:

table Passagers(PassagerId,survived,pclass ,name,age,sex,sibps,parch); 
table Embarquement(Ticket,Fare, Cabin,Embarked,Survived,Name,SibSp,Pclass,PassagerId);
table Titanic(PassagerId,Survived,Pclass ,Name,Age,Sex,Sibps,Parch,Fare,Cabin,Embarked);

Dans un premier temps, pour faire nos tables plusieurs paramètres étaient à prendre en compte tels que les clés primaire,les clés étrangère au sein de notre base de données.
Les differentes relations concernant nos tables étaient également à prendre en compte notamment en ce qui concerne les requetes. 
Dans ce sens, nous avons trouvé qu'il était préférable de créer trois tables différentes dont l'une est la copie conforme du fichier csv. 
Notre base de donnés est alors composé d'une table ayant des caractéristiques sur les passagers. 
Une autre dans laquelle il y a des informations concernant les embarquements 
Nous avons décidez ma camarade et moi de créer ces deux tables pour mettre en avant d'eventuelle relations entre les personnes ayant survecu et leur "situation au sein du bateau" ce qui nous permettrait de relier la table passagers et embarquement afin d'analyser une possible corrélation.

Table Passagers :

La table Passagers est une table qui continent toutes les informations nécessaires concernant les passagers à bord du Titanic. La clé primaire de la table passager est PassagerId car elle permet d’identifier les tuples au sein de la relation. Il est nécessaire de vérifié que chaque tuples soit composé d'une valeur atomique . Elle permet de différencier les éléments de la relation. La clé étrangère de la table Passagers est PassagerId faisant référence à la table Titanic.
La table passagers est donc constitué de 8 attributs et celle ci est composé de 891 lignes(autrement des tuples). 
Le degré de la table est donc de 8 et la cardinalité est de 891. 


Model de donné de la table Passager/Domaine des attributs:

PASSAGER(PassagerId,name,age,sex,sibps,parch,survived) 
dom(PassagerId)=valeurs numériques 
dom(Name)=chaine de caractère 
dom(Age)=float
dom(Sex)=chaine de caractère 
dom(Sibps)=valeur numérique
dom(Parch)=valeur numérique 
dom(Survived)=valeur numérique
dom(Sex)=chaine de caractère

Table Embarquement : 

La table embarquement suit la même logique que la table Passagers dans le sens où celle-ci contient les informations qui concernant indirectement les passagers comme par exemple, la ville dans laquelle ils ont embarqué, le numéro de leurs tickets… 
La clé primaire de cette table est donc l’attribut Name. Name fait office de clé primaire car il différencie à son tour les éléments de la relation. La clé étrangère de la table embarquement est PassagerId faisant référence à la table Titanic.
Chaque numéro de ticket est unique, il permet donc d’identifier les individus.
La table embarquement est donc constitué de 8 attributs et la cardinalité est de 891. 

Model de donné de la table Embarquement/Domaine des attributs: 

Embarquement(Ticket,embarked,cabin,fare,sibps,pclasse,PassagerId)
dom(Ticket)=valeurs numériques 
dom(Embarked)=chaine de caractère
dom(Cabin)=valeurs numériques 
dom(Fare)=float 
dom(Sibps)= valeurs numeriques
dom(Pclasse)=valeurs numériques 
dom(PassagerId)=valeurs numériques


Pour finir , nous avons créer la table TITANIC dans le bus de pourvoir reliée la table passagers et embarquement. Cela nous a aider à créer les clés étrangère.


In [203]:
%%sql

CREATE TABLE TITANIC (PassagerID int PRIMARY KEY,
                      Survived int ,
                      Pclass int , 
                      Name varchar(150),
                      Sex varchar(50),
                      Age float ,SibSp int, 
                      Parch int , 
                      Ticket varchar (150), 
                      Fare float, 
                      Cabin varchar(150), 
                      Embarked varchar (150));

 * postgresql://oceane:***@localhost/postgres
(psycopg2.errors.DuplicateTable) relation "titanic" already exists

[SQL: CREATE TABLE TITANIC (PassagerID int PRIMARY KEY,
                      Survived int ,
                      Pclass int , 
                      Name varchar(150),
                      Sex varchar(50),
                      Age float ,SibSp int, 
                      Parch int , 
                      Ticket varchar (150), 
                      Fare float, 
                      Cabin varchar(150), 
                      Embarked varchar (150));]
(Background on this error at: http://sqlalche.me/e/14/f405)


In [204]:
%%sql 
COPY TITANIC FROM '/Users/oceane/Downloads/titanic_train.csv' DELIMITER','CSV HEADER;

 * postgresql://oceane:***@localhost/postgres


IntegrityError: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "titanic_pkey"
DETAIL:  Key (passagerid)=(1) already exists.
CONTEXT:  COPY titanic, line 2

[SQL: COPY TITANIC FROM '/Users/oceane/Downloads/titanic_train.csv' DELIMITER','CSV HEADER;]
(Background on this error at: http://sqlalche.me/e/14/gkpj)

In [214]:
%%sql

CREATE TABLE passagers (PassagerID integer PRIMARY KEY,
                       Survived integer,
                       Pclass integer,
                       Name varchar(150),
                       Sex varchar(50),
                       Age float,
                       SibSp integer,
                       Parch integer, 
                       foreign key (PassagerID) references TITANIC (PassagerID));

 * postgresql://oceane:***@localhost/postgres
(psycopg2.errors.DuplicateTable) relation "passagers" already exists

[SQL: CREATE TABLE passagers (PassagerID integer PRIMARY KEY,
                       Survived integer,
                       Pclass integer,
                       Name varchar(150),
                       Sex varchar(50),
                       Age float,
                       SibSp integer,
                       Parch integer, 
                       foreign key (PassagerID) references TITANIC (PassagerID));]
(Background on this error at: http://sqlalche.me/e/14/f405)


In [ ]:
# Un message d'erreur apparait (unique violation) 
#car nous ne pouvont pas réacffecter les valeurs alors qu'elles y sont déja.

In [215]:
%%sql 
COPY passagers FROM '/Users/oceane/Downloads/tab_passagers.csv' DELIMITER';'CSV HEADER;

 * postgresql://oceane:***@localhost/postgres


IntegrityError: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "passagers_pkey"
DETAIL:  Key (passagerid)=(1) already exists.
CONTEXT:  COPY passagers, line 2

[SQL: COPY passagers FROM '/Users/oceane/Downloads/tab_passagers.csv' DELIMITER';'CSV HEADER;]
(Background on this error at: http://sqlalche.me/e/14/gkpj)

In [207]:
%%sql
SELECT*FROM passagers;

 * postgresql://oceane:***@localhost/postgres
891 rows affected.


[(1, 0, 3, 'Braund, Mr. Owen Harris', 'male', 22.0, 1, 0),
 (2, 1, 1, 'Cumings, Mrs. John Bradley (Florence Briggs Thayer)', 'female', 38.0, 1, 0),
 (3, 1, 3, 'Heikkinen, Miss. Laina', 'female', 26.0, 0, 0),
 (4, 1, 1, 'Futrelle, Mrs. Jacques Heath (Lily May Peel)', 'female', 35.0, 1, 0),
 (5, 0, 3, 'Allen, Mr. William Henry', 'male', 35.0, 0, 0),
 (6, 0, 3, 'Moran, Mr. James', 'male', None, 0, 0),
 (7, 0, 1, 'McCarthy, Mr. Timothy J', 'male', 54.0, 0, 0),
 (8, 0, 3, 'Palsson, Master. Gosta Leonard', 'male', 2.0, 3, 1),
 (9, 1, 3, 'Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)', 'female', 27.0, 0, 2),
 (10, 1, 2, 'Nasser, Mrs. Nicholas (Adele Achem)', 'female', 14.0, 1, 0),
 (11, 1, 3, 'Sandstrom, Miss. Marguerite Rut', 'female', 4.0, 1, 1),
 (12, 1, 1, 'Bonnell, Miss. Elizabeth', 'female', 58.0, 0, 0),
 (13, 0, 3, 'Saundercock, Mr. William Henry', 'male', 20.0, 0, 0),
 (14, 0, 3, 'Andersson, Mr. Anders Johan', 'male', 39.0, 1, 5),
 (15, 0, 3, 'Vestrom, Miss. Hulda Amanda Adolfina', 'female', 14.0, 0, 0),
 (16, 1, 2, 'Hewlett, Mrs. (Mary D Kingcome) ', 'female', 55.0, 0, 0),
 (17, 0, 3, 'Rice, Master. Eugene', 'male', 2.0, 4, 1),
 (18, 1, 2, 'Williams, Mr. Charles Eugene', 'male', None, 0, 0),
 (19, 0, 3, 'Vander Planke, Mrs. Julius (Emelia Maria Vandemoortele)', 'female', 31.0, 1, 0),
 (20, 1, 3, 'Masselmani, Mrs. Fatima', 'female', None, 0, 0),
 (21, 0, 2, 'Fynney, Mr. Joseph J', 'male', 35.0, 0, 0),
 (22, 1, 2, 'Beesley, Mr. Lawrence', 'male', 34.0, 0, 0),
 (23, 1, 3, 'McGowan, Miss. Anna "Annie"', 'female', 15.0, 0, 0),
 (24, 1, 1, 'Sloper, Mr. William Thompson', 'male', 28.0, 0, 0),
 (25, 0, 3, 'Palsson, Miss. Torborg Danira', 'female', 8.0, 3, 1),
 (26, 1, 3, 'Asplund, Mrs. Carl Oscar (Selma Augusta Emilia Johansson)', 'female', 38.0, 1, 5),
 (27, 0, 3, 'Emir, Mr. Farred Chehab', 'male', None, 0, 0),
 (28, 0, 1, 'Fortune, Mr. Charles Alexander', 'male', 19.0, 3, 2),
 (29, 1, 3, 'O\'Dwyer, Miss. Ellen "Nellie"', 'female', None, 0, 0),
 (30, 0, 3, 'Todoroff, Mr. Lalio', 'male', None, 0, 0),
 (31, 0, 1, 'Uruchurtu, Don. Manuel E', 'male', 40.0, 0, 0),
 (32, 1, 1, 'Spencer, Mrs. William Augustus (Marie Eugenie)', 'female', None, 1, 0),
 (33, 1, 3, 'Glynn, Miss. Mary Agatha', 'female', None, 0, 0),
 (34, 0, 2, 'Wheadon, Mr. Edward H', 'male', 66.0, 0, 0),
 (35, 0, 1, 'Meyer, Mr. Edgar Joseph', 'male', 28.0, 1, 0),
 (36, 0, 1, 'Holverson, Mr. Alexander Oskar', 'male', 42.0, 1, 0),
 (37, 1, 3, 'Mamee, Mr. Hanna', 'male', None, 0, 0),
 (38, 0, 3, 'Cann, Mr. Ernest Charles', 'male', 21.0, 0, 0),
 (39, 0, 3, 'Vander Planke, Miss. Augusta Maria', 'female', 18.0, 2, 0),
 (40, 1, 3, 'Nicola-Yarred, Miss. Jamila', 'female', 14.0, 1, 0),
 (41, 0, 3, 'Ahlin, Mrs. Johan (Johanna Persdotter Larsson)', 'female', 40.0, 1, 0),
 (42, 0, 2, 'Turpin, Mrs. William John Robert (Dorothy Ann Wonnacott)', 'female', 27.0, 1, 0),
 (43, 0, 3, 'Kraeff, Mr. Theodor', 'male', None, 0, 0),
 (44, 1, 2, 'Laroche, Miss. Simonne Marie Anne Andree', 'female', 3.0, 1, 2),
 (45, 1, 3, 'Devaney, Miss. Margaret Delia', 'female', 19.0, 0, 0),
 (46, 0, 3, 'Rogers, Mr. William John', 'male', None, 0, 0),
 (47, 0, 3, 'Lennon, Mr. Denis', 'male', None, 1, 0),
 (48, 1, 3, "O'Driscoll, Miss. Bridget", 'female', None, 0, 0),
 (49, 0, 3, 'Samaan, Mr. Youssef', 'male', None, 2, 0),
 (50, 0, 3, 'Arnold-Franchi, Mrs. Josef (Josefine Franchi)', 'female', 18.0, 1, 0),
 (51, 0, 3, 'Panula, Master. Juha Niilo', 'male', 7.0, 4, 1),
 (52, 0, 3, 'Nosworthy, Mr. Richard Cater', 'male', 21.0, 0, 0),
 (53, 1, 1, 'Harper, Mrs. Henry Sleeper (Myna Haxtun)', 'female', 49.0, 1, 0),
 (54, 1, 2, 'Faunthorpe, Mrs. Lizzie (Elizabeth Anne Wilkinson)', 'female', 29.0, 1, 0),
 (55, 0, 1, 'Ostby, Mr. Engelhart Cornelius', 'male', 65.0, 0, 1),
 (56, 1, 1, 'Woolner, Mr. Hugh', 'male', None, 0, 0),
 (57, 1, 2, 'Rugg, Miss. Emily', 'female', 21.0, 0, 0),
 (58, 0, 3, 'Novel, Mr. Mansouer', 'male', 28.5, 0, 0),
 (59, 1, 2, 'West, Miss. Constance Mirium', 'female', 5.0, 1, 2),
 (60, 0, 3, 'Goodwin, Ma

In [208]:
%%sql
DROP TABLE IF EXISTS embarquement;
CREATE TABLE embarquement(Ticket varchar(50)  , 
                          Fare float,
                          Cabin varchar(150),
                          Embarked varchar(50),
                          Survived integer,
                          Name varchar(150) PRIMARY KEY,
                          SibSp integer,
                          Pclass integer,PassagerID integer , 
                          foreign key (PassagerID) references TITANIC (PassagerID));

 * postgresql://oceane:***@localhost/postgres
Done.
Done.


ResourceClosedError: This result object does not return rows. It has been closed automatically.

In [209]:
%%sql 
COPY embarquement FROM '/Users/oceane/Downloads/table_embarquement-2.csv' DELIMITER';'CSV HEADER;

 * postgresql://oceane:***@localhost/postgres
891 rows affected.


ResourceClosedError: This result object does not return rows. It has been closed automatically.

In [210]:
%%sql
SELECT*FROM embarquement

 * postgresql://oceane:***@localhost/postgres
891 rows affected.


[('A/5 21171', 7.25, None, 'S', 0, 'Braund, Mr. Owen Harris', 1, 3, 1),
 ('PC 17599', 71.2833, 'C85', 'C', 1, 'Cumings, Mrs. John Bradley (Florence Briggs Thayer)', 1, 1, 2),
 ('STON/O2. 3101282', 7.925, None, 'S', 1, 'Heikkinen, Miss. Laina', 0, 3, 3),
 ('113803', 53.1, 'C123', 'S', 1, 'Futrelle, Mrs. Jacques Heath (Lily May Peel)', 1, 1, 4),
 ('373450', 8.05, None, 'S', 0, 'Allen, Mr. William Henry', 0, 3, 5),
 ('330877', 8.4583, None, 'Q', 0, 'Moran, Mr. James', 0, 3, 6),
 ('17463', 51.8625, 'E46', 'S', 0, 'McCarthy, Mr. Timothy J', 0, 1, 7),
 ('349909', 21.075, None, 'S', 0, 'Palsson, Master. Gosta Leonard', 3, 3, 8),
 ('347742', 11.1333, None, 'S', 1, 'Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)', 0, 3, 9),
 ('237736', 30.0708, None, 'C', 1, 'Nasser, Mrs. Nicholas (Adele Achem)', 1, 2, 10),
 ('PP 9549', 16.7, 'G6', 'S', 1, 'Sandstrom, Miss. Marguerite Rut', 1, 3, 11),
 ('113783', 26.55, 'C103', 'S', 1, 'Bonnell, Miss. Elizabeth', 0, 1, 12),
 ('A/5. 2151', 8.05, None, 'S', 0, 'Saundercock, Mr. William Henry', 0, 3, 13),
 ('347082', 31.275, None, 'S', 0, 'Andersson, Mr. Anders Johan', 1, 3, 14),
 ('350406', 7.8542, None, 'S', 0, 'Vestrom, Miss. Hulda Amanda Adolfina', 0, 3, 15),
 ('248706', 16.0, None, 'S', 1, 'Hewlett, Mrs. (Mary D Kingcome) ', 0, 2, 16),
 ('382652', 29.125, None, 'Q', 0, 'Rice, Master. Eugene', 4, 3, 17),
 ('244373', 13.0, None, 'S', 1, 'Williams, Mr. Charles Eugene', 0, 2, 18),
 ('345763', 18.0, None, 'S', 0, 'Vander Planke, Mrs. Julius (Emelia Maria Vandemoortele)', 1, 3, 19),
 ('2649', 7.225, None, 'C', 1, 'Masselmani, Mrs. Fatima', 0, 3, 20),
 ('239865', 26.0, None, 'S', 0, 'Fynney, Mr. Joseph J', 0, 2, 21),
 ('248698', 13.0, 'D56', 'S', 1, 'Beesley, Mr. Lawrence', 0, 2, 22),
 ('330923', 8.0292, None, 'Q', 1, 'McGowan, Miss. Anna "Annie"', 0, 3, 23),
 ('113788', 35.5, 'A6', 'S', 1, 'Sloper, Mr. William Thompson', 0, 1, 24),
 ('349909', 21.075, None, 'S', 0, 'Palsson, Miss. Torborg Danira', 3, 3, 25),
 ('347077', 31.3875, None, 'S', 1, 'Asplund, Mrs. Carl Oscar (Selma Augusta Emilia Johansson)', 1, 3, 26),
 ('2631', 7.225, None, 'C', 0, 'Emir, Mr. Farred Chehab', 0, 3, 27),
 ('19950', 263.0, 'C23 C25 C27', 'S', 0, 'Fortune, Mr. Charles Alexander', 3, 1, 28),
 ('330959', 7.8792, None, 'Q', 1, 'O\'Dwyer, Miss. Ellen "Nellie"', 0, 3, 29),
 ('349216', 7.8958, None, 'S', 0, 'Todoroff, Mr. Lalio', 0, 3, 30),
 ('PC 17601', 27.7208, None, 'C', 0, 'Uruchurtu, Don. Manuel E', 0, 1, 31),
 ('PC 17569', 146.5208, 'B78', 'C', 1, 'Spencer, Mrs. William Augustus (Marie Eugenie)', 1, 1, 32),
 ('335677', 7.75, None, 'Q', 1, 'Glynn, Miss. Mary Agatha', 0, 3, 33),
 ('C.A. 24579', 10.5, None, 'S', 0, 'Wheadon, Mr. Edward H', 0, 2, 34),
 ('PC 17604', 82.1708, None, 'C', 0, 'Meyer, Mr. Edgar Joseph', 1, 1, 35),
 ('113789', 52.0, None, 'S', 0, 'Holverson, Mr. Alexander Oskar', 1, 1, 36),
 ('2677', 7.2292, None, 'C', 1, 'Mamee, Mr. Hanna', 0, 3, 37),
 ('A./5. 2152', 8.05, None, 'S', 0, 'Cann, Mr. Ernest Charles', 0, 3, 38),
 ('345764', 18.0, None, 'S', 0, 'Vander Planke, Miss. Augusta Maria', 2, 3, 39),
 ('2651', 11.2417, None, 'C', 1, 'Nicola-Yarred, Miss. Jamila', 1, 3, 40),
 ('7546', 9.475, None, 'S', 0, 'Ahlin, Mrs. Johan (Johanna Persdotter Larsson)', 1, 3, 41),
 ('11668', 21.0, None, 'S', 0, 'Turpin, Mrs. William John Robert (Dorothy Ann Wonnacott)', 1, 2, 42),
 ('349253', 7.8958, None, 'C', 0, 'Kraeff, Mr. Theodor', 0, 3, 43),
 ('SC/Paris 2123', 41.5792, None, 'C', 1, 'Laroche, Miss. Simonne Marie Anne Andree', 1, 2, 44),
 ('330958', 7.8792, None, 'Q', 1, 'Devaney, Miss. Margaret Delia', 0, 3, 45),
 ('S.C./A.4. 23567', 8.05, None, 'S', 0, 'Rogers, Mr. William John', 0, 3, 46),
 ('370371', 15.5, None, 'Q', 0, 'Lennon, Mr. Denis', 1, 3, 47),
 ('14311', 7.75, None, 'Q', 1, "O'Driscoll, Miss. Bridget", 0, 3, 48),
 ('2662', 21.6792, None, 'C', 0, 'Samaan, Mr. Youssef', 2, 3, 49),
 ('349237', 17.8, None, 'S', 0, 'Arnold-Franchi, Mrs. Josef (Josefine Franchi)', 1, 3, 50),
 ('3101295', 39.6875, None, 'S', 0, 'Panula, Mast

## PHASE 3 ##

Objectifs :
-Créer des requetes afin d'interroger notre base de donnée.
-Répondre aux requetes imposées.

Création de requete afin d'intérroger la base de donnée:

-Afficher les noms des passagers présents à bord du titanic
-Quels sont les noms des personnes n'ayant pas survecu à bord du titanic
-Afficher toutes les informations concernant les numéros de cabines qui n'ont pas été fourni
-Quels sont les passagers qui ont embarqués à Cherbourg

In [212]:
%%sql
SELECT NAME FROM Passagers; 

 * postgresql://oceane:***@localhost/postgres
891 rows affected.


[('Braund, Mr. Owen Harris',),
 ('Cumings, Mrs. John Bradley (Florence Briggs Thayer)',),
 ('Heikkinen, Miss. Laina',),
 ('Futrelle, Mrs. Jacques Heath (Lily May Peel)',),
 ('Allen, Mr. William Henry',),
 ('Moran, Mr. James',),
 ('McCarthy, Mr. Timothy J',),
 ('Palsson, Master. Gosta Leonard',),
 ('Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)',),
 ('Nasser, Mrs. Nicholas (Adele Achem)',),
 ('Sandstrom, Miss. Marguerite Rut',),
 ('Bonnell, Miss. Elizabeth',),
 ('Saundercock, Mr. William Henry',),
 ('Andersson, Mr. Anders Johan',),
 ('Vestrom, Miss. Hulda Amanda Adolfina',),
 ('Hewlett, Mrs. (Mary D Kingcome) ',),
 ('Rice, Master. Eugene',),
 ('Williams, Mr. Charles Eugene',),
 ('Vander Planke, Mrs. Julius (Emelia Maria Vandemoortele)',),
 ('Masselmani, Mrs. Fatima',),
 ('Fynney, Mr. Joseph J',),
 ('Beesley, Mr. Lawrence',),
 ('McGowan, Miss. Anna "Annie"',),
 ('Sloper, Mr. William Thompson',),
 ('Palsson, Miss. Torborg Danira',),
 ('Asplund, Mrs. Carl Oscar (Selma Augusta Emilia Johansson)',),
 ('Emir, Mr. Farred Chehab',),
 ('Fortune, Mr. Charles Alexander',),
 ('O\'Dwyer, Miss. Ellen "Nellie"',),
 ('Todoroff, Mr. Lalio',),
 ('Uruchurtu, Don. Manuel E',),
 ('Spencer, Mrs. William Augustus (Marie Eugenie)',),
 ('Glynn, Miss. Mary Agatha',),
 ('Wheadon, Mr. Edward H',),
 ('Meyer, Mr. Edgar Joseph',),
 ('Holverson, Mr. Alexander Oskar',),
 ('Mamee, Mr. Hanna',),
 ('Cann, Mr. Ernest Charles',),
 ('Vander Planke, Miss. Augusta Maria',),
 ('Nicola-Yarred, Miss. Jamila',),
 ('Ahlin, Mrs. Johan (Johanna Persdotter Larsson)',),
 ('Turpin, Mrs. William John Robert (Dorothy Ann Wonnacott)',),
 ('Kraeff, Mr. Theodor',),
 ('Laroche, Miss. Simonne Marie Anne Andree',),
 ('Devaney, Miss. Margaret Delia',),
 ('Rogers, Mr. William John',),
 ('Lennon, Mr. Denis',),
 ("O'Driscoll, Miss. Bridget",),
 ('Samaan, Mr. Youssef',),
 ('Arnold-Franchi, Mrs. Josef (Josefine Franchi)',),
 ('Panula, Master. Juha Niilo',),
 ('Nosworthy, Mr. Richard Cater',),
 ('Harper, Mrs. Henry Sleeper (Myna Haxtun)',),
 ('Faunthorpe, Mrs. Lizzie (Elizabeth Anne Wilkinson)',),
 ('Ostby, Mr. Engelhart Cornelius',),
 ('Woolner, Mr. Hugh',),
 ('Rugg, Miss. Emily',),
 ('Novel, Mr. Mansouer',),
 ('West, Miss. Constance Mirium',),
 ('Goodwin, Master. William Frederick',),
 ('Sirayanian, Mr. Orsen',),
 ('Icard, Miss. Amelie',),
 ('Harris, Mr. Henry Birkhardt',),
 ('Skoog, Master. Harald',),
 ('Stewart, Mr. Albert A',),
 ('Moubarek, Master. Gerios',),
 ('Nye, Mrs. (Elizabeth Ramell)',),
 ('Crease, Mr. Ernest James',),
 ('Andersson, Miss. Erna Alexandra',),
 ('Kink, Mr. Vincenz',),
 ('Jenkin, Mr. Stephen Curnow',),
 ('Goodwin, Miss. Lillian Amy',),
 ('Hood, Mr. Ambrose Jr',),
 ('Chronopoulos, Mr. Apostolos',),
 ('Bing, Mr. Lee',),
 ('Moen, Mr. Sigurd Hansen',),
 ('Staneff, Mr. Ivan',),
 ('Moutal, Mr. Rahamin Haim',),
 ('Caldwell, Master. Alden Gates',),
 ('Dowdell, Miss. Elizabeth',),
 ('Waelens, Mr. Achille',),
 ('Sheerlinck, Mr. Jan Baptist',),
 ('McDermott, Miss. Brigdet Delia',),
 ('Carrau, Mr. Francisco M',),
 ('Ilett, Miss. Bertha',),
 ('Backstrom, Mrs. Karl Alfred (Maria Mathilda Gustafsson)',),
 ('Ford, Mr. William Neal',),
 ('Slocovski, Mr. Selman Francis',),
 ('Fortune, Miss. Mabel Helen',),
 ('Celotti, Mr. Francesco',),
 ('Christmann, Mr. Emil',),
 ('Andreasson, Mr. Paul Edvin',),
 ('Chaffee, Mr. Herbert Fuller',),
 ('Dean, Mr. Bertram Frank',),
 ('Coxon, Mr. Daniel',),
 ('Shorney, Mr. Charles Joseph',),
 ('Goldschmidt, Mr. George B',),
 ('Greenfield, Mr. William Bertram',),
 ('Doling, Mrs. John T (Ada Julia Bone)',),
 ('Kantor, Mr. Sinai',),
 ('Petranec, Miss. Matilda',),
 ('Petroff, Mr. Pastcho ("Pentcho")',),
 ('White, Mr. Richard Frasar',),
 ('Johansson, Mr. Gustaf Joel',),
 ('Gustafsson, Mr. Anders Vilhelm',),
 ('Mionoff, Mr. Stoytcho',),
 ('Salkjelsvik, Miss. Anna Kristine',),
 ('Moss, Mr. Albert Johan',),
 ('Rekic, Mr. Tido',),
 ('Moran, Miss. Bertha',),
 ('Porter, Mr. Walter Chamberlain',),
 ('Zabour, Miss. Hileni',),
 ('Barton, Mr. David John',),
 ('Jussila

In [211]:
%%sql 
SELECT Name FROM Passagers where Survived='0';

 * postgresql://oceane:***@localhost/postgres
549 rows affected.


[('Braund, Mr. Owen Harris',),
 ('Allen, Mr. William Henry',),
 ('Moran, Mr. James',),
 ('McCarthy, Mr. Timothy J',),
 ('Palsson, Master. Gosta Leonard',),
 ('Saundercock, Mr. William Henry',),
 ('Andersson, Mr. Anders Johan',),
 ('Vestrom, Miss. Hulda Amanda Adolfina',),
 ('Rice, Master. Eugene',),
 ('Vander Planke, Mrs. Julius (Emelia Maria Vandemoortele)',),
 ('Fynney, Mr. Joseph J',),
 ('Palsson, Miss. Torborg Danira',),
 ('Emir, Mr. Farred Chehab',),
 ('Fortune, Mr. Charles Alexander',),
 ('Todoroff, Mr. Lalio',),
 ('Uruchurtu, Don. Manuel E',),
 ('Wheadon, Mr. Edward H',),
 ('Meyer, Mr. Edgar Joseph',),
 ('Holverson, Mr. Alexander Oskar',),
 ('Cann, Mr. Ernest Charles',),
 ('Vander Planke, Miss. Augusta Maria',),
 ('Ahlin, Mrs. Johan (Johanna Persdotter Larsson)',),
 ('Turpin, Mrs. William John Robert (Dorothy Ann Wonnacott)',),
 ('Kraeff, Mr. Theodor',),
 ('Rogers, Mr. William John',),
 ('Lennon, Mr. Denis',),
 ('Samaan, Mr. Youssef',),
 ('Arnold-Franchi, Mrs. Josef (Josefine Franchi)',),
 ('Panula, Master. Juha Niilo',),
 ('Nosworthy, Mr. Richard Cater',),
 ('Ostby, Mr. Engelhart Cornelius',),
 ('Novel, Mr. Mansouer',),
 ('Goodwin, Master. William Frederick',),
 ('Sirayanian, Mr. Orsen',),
 ('Harris, Mr. Henry Birkhardt',),
 ('Skoog, Master. Harald',),
 ('Stewart, Mr. Albert A',),
 ('Crease, Mr. Ernest James',),
 ('Kink, Mr. Vincenz',),
 ('Jenkin, Mr. Stephen Curnow',),
 ('Goodwin, Miss. Lillian Amy',),
 ('Hood, Mr. Ambrose Jr',),
 ('Chronopoulos, Mr. Apostolos',),
 ('Moen, Mr. Sigurd Hansen',),
 ('Staneff, Mr. Ivan',),
 ('Moutal, Mr. Rahamin Haim',),
 ('Waelens, Mr. Achille',),
 ('Carrau, Mr. Francisco M',),
 ('Ford, Mr. William Neal',),
 ('Slocovski, Mr. Selman Francis',),
 ('Celotti, Mr. Francesco',),
 ('Christmann, Mr. Emil',),
 ('Andreasson, Mr. Paul Edvin',),
 ('Chaffee, Mr. Herbert Fuller',),
 ('Dean, Mr. Bertram Frank',),
 ('Coxon, Mr. Daniel',),
 ('Shorney, Mr. Charles Joseph',),
 ('Goldschmidt, Mr. George B',),
 ('Kantor, Mr. Sinai',),
 ('Petranec, Miss. Matilda',),
 ('Petroff, Mr. Pastcho ("Pentcho")',),
 ('White, Mr. Richard Frasar',),
 ('Johansson, Mr. Gustaf Joel',),
 ('Gustafsson, Mr. Anders Vilhelm',),
 ('Mionoff, Mr. Stoytcho',),
 ('Rekic, Mr. Tido',),
 ('Porter, Mr. Walter Chamberlain',),
 ('Zabour, Miss. Hileni',),
 ('Barton, Mr. David John',),
 ('Jussila, Miss. Katriina',),
 ('Attalah, Miss. Malake',),
 ('Pekoniemi, Mr. Edvard',),
 ('Connors, Mr. Patrick',),
 ('Turpin, Mr. William John Robert',),
 ('Baxter, Mr. Quigg Edmond',),
 ('Andersson, Miss. Ellis Anna Maria',),
 ('Hickman, Mr. Stanley George',),
 ('Moore, Mr. Leonard Charles',),
 ('Nasser, Mr. Nicholas',),
 ('White, Mr. Percival Wayland',),
 ('McMahon, Mr. Martin',),
 ('Ekstrom, Mr. Johan',),
 ('Drazenoic, Mr. Jozef',),
 ('Coelho, Mr. Domingos Fernandeo',),
 ('Robins, Mrs. Alexander A (Grace Charity Laury)',),
 ('Sobey, Mr. Samuel James Hayden',),
 ('Richard, Mr. Emile',),
 ('Futrelle, Mr. Jacques Heath',),
 ('Osen, Mr. Olaf Elon',),
 ('Giglio, Mr. Victor',),
 ('Boulos, Mrs. Joseph (Sultana)',),
 ('Burke, Mr. Jeremiah',),
 ('Andrew, Mr. Edgardo Samuel',),
 ('Nicholls, Mr. Joseph Charles',),
 ('Ford, Miss. Robina Maggie "Ruby"',),
 ('Navratil, Mr. Michel ("Louis M Hoffman")',),
 ('Byles, Rev. Thomas Roussel Davids',),
 ('Bateman, Rev. Robert James',),
 ('Meo, Mr. Alfonzo',),
 ('van Billiard, Mr. Austin Blyler',),
 ('Olsen, Mr. Ole Martin',),
 ('Williams, Mr. Charles Duane',),
 ('Corn, Mr. Harry',),
 ('Smiljanic, Mr. Mile',),
 ('Sage, Master. Thomas Henry',),
 ('Cribb, Mr. John Hatfield',),
 ('Bengtsson, Mr. John Viktor',),
 ('Calic, Mr. Jovo',),
 ('Panula, Master. Eino Viljami',),
 ('Skoog, Mrs. William (Anna Bernhardina Karlsson)',),
 ('Baumann, Mr. John D',),
 ('Ling, Mr. Lee',),
 ('Van der hoef, Mr. Wyckoff',),
 ('Rice, Master. Arthur',),
 ('Sivola, Mr. Antti Wilhelm',),
 ('Smith, Mr. James Clinch',),
 ('Klasen, Mr. Klas Albin',),
 ('Lefebre, Master. Henry Forbes',),
 ('Isham, Miss. Ann Elizabeth',),
 ('Hale, Mr. Reginald',),
 ('Leonard, Mr

In [145]:
%%sql
SELECT*FROM Embarquement where Cabin is null;

 * postgresql://oceane:***@localhost/postgres
687 rows affected.


[('A/5 21171', 7.25, None, 'S', 0, 'Braund, Mr. Owen Harris', 1, 3, 1),
 ('STON/O2. 3101282', 7.925, None, 'S', 1, 'Heikkinen, Miss. Laina', 0, 3, 3),
 ('373450', 8.05, None, 'S', 0, 'Allen, Mr. William Henry', 0, 3, 5),
 ('330877', 8.4583, None, 'Q', 0, 'Moran, Mr. James', 0, 3, 6),
 ('349909', 21.075, None, 'S', 0, 'Palsson, Master. Gosta Leonard', 3, 3, 8),
 ('347742', 11.1333, None, 'S', 1, 'Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)', 0, 3, 9),
 ('237736', 30.0708, None, 'C', 1, 'Nasser, Mrs. Nicholas (Adele Achem)', 1, 2, 10),
 ('A/5. 2151', 8.05, None, 'S', 0, 'Saundercock, Mr. William Henry', 0, 3, 13),
 ('347082', 31.275, None, 'S', 0, 'Andersson, Mr. Anders Johan', 1, 3, 14),
 ('350406', 7.8542, None, 'S', 0, 'Vestrom, Miss. Hulda Amanda Adolfina', 0, 3, 15),
 ('248706', 16.0, None, 'S', 1, 'Hewlett, Mrs. (Mary D Kingcome) ', 0, 2, 16),
 ('382652', 29.125, None, 'Q', 0, 'Rice, Master. Eugene', 4, 3, 17),
 ('244373', 13.0, None, 'S', 1, 'Williams, Mr. Charles Eugene', 0, 2, 18),
 ('345763', 18.0, None, 'S', 0, 'Vander Planke, Mrs. Julius (Emelia Maria Vandemoortele)', 1, 3, 19),
 ('2649', 7.225, None, 'C', 1, 'Masselmani, Mrs. Fatima', 0, 3, 20),
 ('239865', 26.0, None, 'S', 0, 'Fynney, Mr. Joseph J', 0, 2, 21),
 ('330923', 8.0292, None, 'Q', 1, 'McGowan, Miss. Anna "Annie"', 0, 3, 23),
 ('349909', 21.075, None, 'S', 0, 'Palsson, Miss. Torborg Danira', 3, 3, 25),
 ('347077', 31.3875, None, 'S', 1, 'Asplund, Mrs. Carl Oscar (Selma Augusta Emilia Johansson)', 1, 3, 26),
 ('2631', 7.225, None, 'C', 0, 'Emir, Mr. Farred Chehab', 0, 3, 27),
 ('330959', 7.8792, None, 'Q', 1, 'O\'Dwyer, Miss. Ellen "Nellie"', 0, 3, 29),
 ('349216', 7.8958, None, 'S', 0, 'Todoroff, Mr. Lalio', 0, 3, 30),
 ('PC 17601', 27.7208, None, 'C', 0, 'Uruchurtu, Don. Manuel E', 0, 1, 31),
 ('335677', 7.75, None, 'Q', 1, 'Glynn, Miss. Mary Agatha', 0, 3, 33),
 ('C.A. 24579', 10.5, None, 'S', 0, 'Wheadon, Mr. Edward H', 0, 2, 34),
 ('PC 17604', 82.1708, None, 'C', 0, 'Meyer, Mr. Edgar Joseph', 1, 1, 35),
 ('113789', 52.0, None, 'S', 0, 'Holverson, Mr. Alexander Oskar', 1, 1, 36),
 ('2677', 7.2292, None, 'C', 1, 'Mamee, Mr. Hanna', 0, 3, 37),
 ('A./5. 2152', 8.05, None, 'S', 0, 'Cann, Mr. Ernest Charles', 0, 3, 38),
 ('345764', 18.0, None, 'S', 0, 'Vander Planke, Miss. Augusta Maria', 2, 3, 39),
 ('2651', 11.2417, None, 'C', 1, 'Nicola-Yarred, Miss. Jamila', 1, 3, 40),
 ('7546', 9.475, None, 'S', 0, 'Ahlin, Mrs. Johan (Johanna Persdotter Larsson)', 1, 3, 41),
 ('11668', 21.0, None, 'S', 0, 'Turpin, Mrs. William John Robert (Dorothy Ann Wonnacott)', 1, 2, 42),
 ('349253', 7.8958, None, 'C', 0, 'Kraeff, Mr. Theodor', 0, 3, 43),
 ('SC/Paris 2123', 41.5792, None, 'C', 1, 'Laroche, Miss. Simonne Marie Anne Andree', 1, 2, 44),
 ('330958', 7.8792, None, 'Q', 1, 'Devaney, Miss. Margaret Delia', 0, 3, 45),
 ('S.C./A.4. 23567', 8.05, None, 'S', 0, 'Rogers, Mr. William John', 0, 3, 46),
 ('370371', 15.5, None, 'Q', 0, 'Lennon, Mr. Denis', 1, 3, 47),
 ('14311', 7.75, None, 'Q', 1, "O'Driscoll, Miss. Bridget", 0, 3, 48),
 ('2662', 21.6792, None, 'C', 0, 'Samaan, Mr. Youssef', 2, 3, 49),
 ('349237', 17.8, None, 'S', 0, 'Arnold-Franchi, Mrs. Josef (Josefine Franchi)', 1, 3, 50),
 ('3101295', 39.6875, None, 'S', 0, 'Panula, Master. Juha Niilo', 4, 3, 51),
 ('A/4. 39886', 7.8, None, 'S', 0, 'Nosworthy, Mr. Richard Cater', 0, 3, 52),
 ('2926', 26.0, None, 'S', 1, 'Faunthorpe, Mrs. Lizzie (Elizabeth Anne Wilkinson)', 1, 2, 54),
 ('C.A. 31026', 10.5, None, 'S', 1, 'Rugg, Miss. Emily', 0, 2, 57),
 ('2697', 7.2292, None, 'C', 0, 'Novel, Mr. Mansouer', 0, 3, 58),
 ('C.A. 34651', 27.75, None, 'S', 1, 'West, Miss. Constance Mirium', 1, 2, 59),
 ('CA 2144', 46.9, None, 'S', 0, 'Goodwin, Master. William Frederick', 5, 3, 60),
 ('2669', 7.2292, None, 'C', 0, 'Sirayanian, Mr. Orsen', 0, 3, 61),
 ('347088', 27.9, None, 'S', 0, 'Skoog, Master. Harald', 3, 3, 64),
 ('PC 17605', 27.7208, None, 'C', 0, 'Stewart, Mr. Albert A', 0, 1, 65),
 ('2661', 15.2458, None, 'C', 1, 'Moubarek, 

In [146]:
%%sql
SELECT Name FROM Embarquement
where embarked='C';



 * postgresql://oceane:***@localhost/postgres
168 rows affected.


[('Cumings, Mrs. John Bradley (Florence Briggs Thayer)',),
 ('Nasser, Mrs. Nicholas (Adele Achem)',),
 ('Masselmani, Mrs. Fatima',),
 ('Emir, Mr. Farred Chehab',),
 ('Uruchurtu, Don. Manuel E',),
 ('Spencer, Mrs. William Augustus (Marie Eugenie)',),
 ('Meyer, Mr. Edgar Joseph',),
 ('Mamee, Mr. Hanna',),
 ('Nicola-Yarred, Miss. Jamila',),
 ('Kraeff, Mr. Theodor',),
 ('Laroche, Miss. Simonne Marie Anne Andree',),
 ('Samaan, Mr. Youssef',),
 ('Harper, Mrs. Henry Sleeper (Myna Haxtun)',),
 ('Ostby, Mr. Engelhart Cornelius',),
 ('Novel, Mr. Mansouer',),
 ('Sirayanian, Mr. Orsen',),
 ('Stewart, Mr. Albert A',),
 ('Moubarek, Master. Gerios',),
 ('Chronopoulos, Mr. Apostolos',),
 ('Goldschmidt, Mr. George B',),
 ('Greenfield, Mr. William Bertram',),
 ('Zabour, Miss. Hileni',),
 ('Attalah, Miss. Malake',),
 ('Baxter, Mr. Quigg Edmond',),
 ('Nasser, Mr. Nicholas',),
 ('Nicola-Yarred, Master. Elias',),
 ('Peter, Miss. Anna',),
 ('Drazenoic, Mr. Jozef',),
 ('Richard, Mr. Emile',),
 ('Giglio, Mr. Victor',),
 ('Boulos, Mrs. Joseph (Sultana)',),
 ('Williams, Mr. Charles Duane',),
 ('Smith, Mr. James Clinch',),
 ('Isham, Miss. Ann Elizabeth',),
 ('Pernot, Mr. Rene',),
 ('Brown, Mrs. James Joseph (Margaret Tobin)',),
 ('Lurette, Miss. Elise',),
 ('Youseff, Mr. Gerious',),
 ('Albimona, Mr. Nassef Cassem',),
 ('Blank, Mr. Henry',),
 ('Newell, Miss. Madeleine',),
 ('Bazzani, Miss. Albina',),
 ('Zabour, Miss. Thamine',),
 ('Attalah, Mr. Sleiman',),
 ('Touma, Mrs. Darwis (Hanne Youssef Razi)',),
 ('Thorne, Mrs. Gertrude Maybelle',),
 ('Ward, Miss. Anna',),
 ('Natsch, Mr. Charles H',),
 ('Stankovic, Mr. Ivan',),
 ('Bishop, Mrs. Dickinson H (Helen Walton)',),
 ('Levy, Mr. Rene Jacques',),
 ('Lewy, Mr. Ervin G',),
 ('Hanna, Mr. Mansour',),
 ('Baxter, Mrs. James (Helene DeLaudeniere Chaput)',),
 ('Fleming, Miss. Margaret',),
 ('Penasco y Castellana, Mrs. Victor de Satode (Maria Josefa Perez de Soto y Vallejo)',),
 ('Abelson, Mr. Samuel',),
 ('Francatelli, Miss. Laura Mabel',),
 ('Hays, Miss. Margaret Bechstein',),
 ('Ryerson, Miss. Emily Borie',),
 ('Spedden, Mrs. Frederic Oakley (Margaretta Corning Stone)',),
 ('Young, Miss. Marie Grice',),
 ('Hippach, Miss. Jean Gertrude',),
 ('Burns, Miss. Elizabeth Margaret',),
 ('Elias, Mr. Tannous',),
 ('Yousif, Mr. Wazli',),
 ('del Carlo, Mr. Sebastiano',),
 ('Barbara, Mrs. (Catherine David)',),
 ('Warren, Mrs. Frank Manley (Anna Sophia Atkinson)',),
 ('Moussa, Mrs. (Mantoura Boulos)',),
 ('Aubart, Mme. Leontine Pauline',),
 ('Harder, Mr. George Achilles',),
 ('Ringhini, Mr. Sante',),
 ('Meyer, Mrs. Edgar Joseph (Leila Saks)',),
 ('Widener, Mr. Harry Elkins',),
 ('Betros, Mr. Tannous',),
 ('Bidois, Miss. Rosalie',),
 ('Nakid, Miss. Maria ("Mary")',),
 ('Lehmann, Miss. Bertha',),
 ('Newell, Miss. Marjorie',),
 ('Gheorgheff, Mr. Stanio',),
 ('Baclini, Miss. Marie Catherine',),
 ('Foreman, Mr. Benjamin Laventall',),
 ('Goldenberg, Mr. Samuel L',),
 ('Jalsevac, Mr. Ivan',),
 ('Baclini, Miss. Helene Barbara',),
 ('Jerwan, Mrs. Amin S (Marie Marthe Thuillard)',),
 ('Bishop, Mr. Dickinson H',),
 ('Kent, Mr. Edward Austin',),
 ('Artagaveytia, Mr. Ramon',),
 ('Yousseff, Mr. Gerious',),
 ('Eustis, Miss. Elizabeth Mussey',),
 ('Penasco y Castellana, Mr. Victor de Satode',),
 ('Rothschild, Mrs. Martin (Elizabeth L. Barrett)',),
 ('Lahoud, Mr. Sarkis',),
 ('Hippach, Mrs. Louis Albert (Ida Sophia Fischer)',),
 ('Kassem, Mr. Fared',),
 ('Toufik, Mr. Nakli',),
 ('Elias, Mr. Joseph Jr',),
 ('Peter, Mrs. Catherine (Catherine Rizk)',),
 ('LeRoy, Miss. Bertha',),
 ('Frolicher, Miss. Hedwig Margaritha',),
 ('Douglas, Mr. Walter Donald',),
 ('Padro y Manent, Mr. Julian',),
 ('Thayer, Mr. John Borland Jr',),
 ('Leeni, Mr. Fahim ("Philip Zenni")',),
 ('Duff Gordon, Lady. (Lucille Christiana Sutherland) ("Mrs Morgan")',),
 ('Robbins, Mr. Victor',),
 ('Doharr, Mr. Tannous',),
 ('Caram, Mrs. Joseph (Maria Elias)',),
 ('Thayer, Mrs. John Borland (Marian Longstreth Morris)',),
 ('Ross, Mr. John Hugo',),
 ('Paulner, Mr. Uscher',),
 ('Frolich

## requete sql ##

requete 1

In [42]:
#Nous utilisons ici la fonction count afin qu'on puisse compter le nombre de classe . 
#De surcroit , nous ajoutons la clause distinct qui permet d'eviter les doublons.

In [147]:
%%sql
SELECT COUNT(DISTINCT pclass ) from embarquement;


 * postgresql://oceane:***@localhost/postgres
1 rows affected.


[(3,)]

requete 2

In [44]:
#La fonction count nous permet de compter les différents informations par classe
#Le GROUP BY est utile ici car il permet de regrouper les sous-ensembles appartenant à un seul tuple

In [148]:
%%sql

Select pclass , COUNT(distinct PassagerId) from passagers GROUP BY pclass;

 * postgresql://oceane:***@localhost/postgres
3 rows affected.


[(1, 216), (2, 184), (3, 491)]

requete 3

In [46]:
# Utilisation du count 
#Utilisation du GROUP BY 

In [213]:
%%sql 
select pclass , sex,  COUNT (*) from passagers where sex='male' or sex='female' GROUP BY pclass ,sex; 

 * postgresql://oceane:***@localhost/postgres
6 rows affected.


[(1, 'female', 94),
 (1, 'male', 122),
 (2, 'male', 108),
 (2, 'female', 76),
 (3, 'female', 144),
 (3, 'male', 347)]

In [48]:
# Nous utilisons ici la fonction qui permet de compter dans un attributs les différentes caractéristiques qu'on lui 
#passe en paramètre. Vu que nous cherchons les passagers ayant survecu = 1 et les passagers n'ayant pas survécu = 0
# Pour calculer le pourcentage nous avons multiplé par 100 pour faire un pourcentage.


In [48]:
# Nous utilisons ici la fonction qui permet de compter dans un attributs les différentes caractéristiques qu'on lui 
#passe en paramètre. Vu que nous cherchons les passagers ayant survecu = 1 et les passagers n'ayant pas survécu = 0
# Pour calculer le pourcentage nous avons multiplé par 100 pour faire un pourcentage.


Requete 4

In [150]:
%%sql 
select survived , COUNT(*) from passagers where survived ='0' or survived ='1' GROUP BY survived;

 * postgresql://oceane:***@localhost/postgres
2 rows affected.


[(0, 549), (1, 342)]

In [151]:
%%sql

select survived , (100 * COUNT(*))  / (select COUNT(*) from passagers) as pourcentage
from passagers 

group by survived ;

 * postgresql://oceane:***@localhost/postgres
2 rows affected.


[(0, 61), (1, 38)]

In [ ]:
#Nous avons crée une vues pour avoir une version simplifié de la table .


requete 5

In [194]:
%%sql

Create view survivant as 
select  survived , pclass , COUNT(*)
from passagers 
where survived ='0' or survived ='1' 
GROUP BY  survived , pclass;


 * postgresql://oceane:***@localhost/postgres
(psycopg2.errors.DuplicateTable) relation "survivant" already exists

[SQL: Create view survivant as 
select  survived , pclass , COUNT(*)
from passagers 
where survived ='0' or survived ='1' 
GROUP BY  survived , pclass;]
(Background on this error at: http://sqlalche.me/e/14/f405)


In [195]:
%%sql

select* from survivant;

 * postgresql://oceane:***@localhost/postgres
6 rows affected.


[(0, 3, 372), (0, 1, 80), (0, 2, 97), (1, 3, 119), (1, 2, 87), (1, 1, 136)]

In [ ]:
#Nous avons crée une vues pour avoir une version simplifié de la table . (table intermédiraire)

requete 6

In [196]:
%%sql

CREATE view SurvivantS as
select survived , sex , COUNT(*) 
from passagers 
where survived ='0' or survived ='1'
GROUP BY survived , sex;



 * postgresql://oceane:***@localhost/postgres
(psycopg2.errors.DuplicateTable) relation "survivants" already exists

[SQL: CREATE view SurvivantS as
select survived , sex , COUNT(*) 
from passagers 
where survived ='0' or survived ='1'
GROUP BY survived , sex;]
(Background on this error at: http://sqlalche.me/e/14/f405)


In [197]:
%%sql
select * from SurvivantS;

 * postgresql://oceane:***@localhost/postgres
4 rows affected.


[(0, 'female', 81), (1, 'female', 233), (1, 'male', 109), (0, 'male', 468)]